### PR TITLE
Frontend dependency management part II

### PR DESCRIPTION
**Pull request by elebitzero**
Part II for #2423.
Partial fix for #3958 (Only suggest package is using minified JavaScript now).

Changes proposed in this pull request:
- Added postinstall script.
- Updated package.json.
- Removed unneeded files in externals.
- Changed file paths from externals to 3rdparty.
- Updated refine.bat and refine scripts to call npm install for the build.
- Updated developer documentation
- imgareaselect, suggest (4.3a), and jquery-ui remain in externals.

### Note
- Verified package contents of Windows zip files and confirmed refine.bat run works without errors.
- Linux tested and confirmed to work.
- Mac installer confirmed to build on Linux but not tested.

### CODE REVIEW

**Review by wetneb — COMMENTED** [review]
Very nice, thank you!

We should update our technical documentation to indicate that `npm` is now required to build the project. This could be done in the same PR since the docs are also stored in this repository (for now): this could go in `docs/docs/technical-reference/build-test-run.md`.

Also, concerning the `dependencies.json` file, I did not anticipate that it would be required, it is a tiny bit sad because it will not be very convenient to maintain that alongside the `package.json`, but I guess it is a necessary hurdle as long as we are not using a proper pipeline with (for instance) webpack. So it feels fine for now :)

**Comment by elebitzero** [comment]
> Also, concerning the `dependencies.json` file, I did not anticipate that it would be required, it is a tiny bit sad because it will not be very convenient to maintain that alongside the `package.json`,

Yes, webpack would be great, but that is a bigger change, and it could have some issues with this Jquery app that is allowing UI extensions. Webpack works great with SPA apps like Angular and React.  I will take a look at what is involved with using webpack after the webapp is using package.json for dependencies.

**Comment by wetneb** [comment]
Fabulous, thank you!
Maybe I am missing something but it looks like the `refine` and `refine.bat` scripts has not been updated yet to call `npm`, right? If you intend to do this in a follow-up PR then the documentation update was perhaps premature (but fine to leave it like this of course).

**Comment by elebitzero** [comment]
> Maybe I am missing something but it looks like the `refine` and `refine.bat` scripts has not been updated yet to call `npm`, right?

Yes, I was going to do a part III, but I just went ahead and added the remaining changes to this PR.

**Comment by elebitzero** [comment]
The installer now works for Windows, but still needs to be tested on Linux and Mac.

**Comment by elebitzero** [comment]
The installer is now tested on Linux and confirmed to work, just Mac remains.

File: refine [hunk 1]
```diff
@@ -585,6 +590,7 @@ run() {
         IS_JAR=`ls $REFINE_LIB_DIR | grep openrefine`
         if [ -z "$IS_JAR" ] ; then
             mvn_prepare
+            build_prepare
```
Comment (wetneb) [inline refine:593]: I have added this to make sure we can run `./refine` just after cloning the repository (implicitly building before launching) as this is supported on Linux.

**Review by wetneb — APPROVED** [review]

**Pull request merged**